experiments/conductor/cmd/runner/crd_generator_commands.go

```diff
@@ -214,119 +207,77 @@ func generateTypesAndMapper(opts *RunnerOptions, branch Branch) {
 				"--service", branch.Package,
 				"--api-version", fmt.Sprintf("%s.cnrm.cloud.google.com/v1alpha1", branch.Group),
 			},
-			WorkDir:    controllerBuilderDir,
+			WorkDir:    filepath.Join(opts.branchRepoDir, "dev", "tools", "controllerbuilder"),
 			MaxRetries: 2,
 		}
 		_, _, err := executeCommand(opts, cfg)
 		if err != nil {
-			log.Fatal(err)
+			return affectedPaths, fmt.Errorf("failed to generate mapper: %v", err)
 		}
 
-		// Run goimports
-		cfg = CommandConfig{
-			Name: "Format generated code",
-			Cmd:  "go",
-			Args: []string{
-				"run", "-mod=readonly",
-				"golang.org/x/tools/cmd/goimports@latest",
-				"-w",
-				mapperDir,
-			},
-			WorkDir:    workDir,
-			MaxRetries: 2,
-		}
-		_, _, err = executeCommand(opts, cfg)
-		if err != nil {
-			log.Fatal(err)
-		}
-		gitAdd(ctx, workDir, mapperDir)
-		hasChange = true
+		affectedPaths = append(affectedPaths, mapperDirPathRelative)
+		generatedCode = true
 	} else {
-		log.Printf("SKIPPING generating mappers, %s already exists", mapperDir)
+		log.Printf("SKIPPING generating mappers, %s already exists", mapperDirPathRelative)
 	}
 
-	// Commit the changes
-	if hasChange {
-		gitCommit(ctx, workDir, fmt.Sprintf("Generated types and mapper for %s", branch.Kind))
-	} else {
-		log.Printf("SKIPPING git commit, no new changes for %s", branch.Name)
+	if !generatedCode {
+		log.Printf("No new changes for %s", branch.Name)
 	}
-}
-
-func generateCRD(opts *RunnerOptions, branch Branch) {
-	ctx := context.TODO()
 
-	close := setLoggingWriter(opts, branch)
-	defer close()
-	workDir := opts.branchRepoDir
+	return affectedPaths, nil
+}
 
-	checkoutBranch(ctx, branch, workDir)
+func generateCRD(ctx context.Context, opts *RunnerOptions, branch Branch) ([]string, error) {
+	affectedPaths := []string{"config/crds/resources/"}
 
 	// Generate CRDs
 	cfg := CommandConfig{
 		Name:       "Generate CRDs",
-		Cmd:        filepath.Join(workDir, "dev", "tasks", "generate-crds"),
-		WorkDir:    workDir,
+		Cmd:        filepath.Join(opts.branchRepoDir, "dev", "tasks", "generate-crds"),
+		WorkDir:    opts.branchRepoDir,
 		MaxRetries: 1,
 	}
-	_, _, err := executeCommand(opts, cfg)
-	if err != nil {
-		log.Fatal(err)
-	}
-
-	// Stage the changed files
-	gitAdd(ctx, workDir, "config/crds/resources/")
 
-	// Commit the changes
-	gitCommit(ctx, workDir, fmt.Sprintf("Generated CRD for %s", branch.Kind))
+	_, _, err := executeCommand(opts, cfg)
+	return affectedPaths, err
```

Comment:
Add `apiDirPathRelative` to affectedPaths (for the deepcopy file).

Reply:
done.